cedar-policy/src/api.rs

```diff
@@ -1294,24 +1300,24 @@ impl PolicySet {
             .into_iter()
             .map(|(key, value)| (key.into(), value.0))
             .collect();
-        self.ast
+        let linked_ast = self
+            .ast
             .link(
                 template_id.0.clone(),
                 new_id.0.clone(),
                 unwrapped_vals.clone(),
             )
             .map_err(PolicySetError::LinkingError)?;
-        let linked_ast = self
-            .ast
-            .get(&new_id.0)
-            .expect("ast.link() didn't fail above, so this shouldn't fail");
         let linked_lossless = self
             .templates
             .get(&template_id)
-            .expect("ast.link() didn't fail above, so this shouldn't fail")
+            .ok_or(PolicySetError::ExpectedTemplate)?
```

Comment:
can we document why this is the right error?  Looking at these couple lines in isolation, it looks like this could fire when the template simply does not exist, which is a different error.  The code is correct but a comment would be helpful for whoever looks at this next